_search-plugins/search-pipelines/rerank-processor.md

```diff
@@ -110,9 +117,166 @@ POST /_search?search_pipeline=rerank_pipeline
+
+### The `by_field` rerank type
+
+The `by_field` rerank type is designed to operate using a specified field within your document. This is useful if a model has already ran and produced a numerical score in your document(s), or if a previous search response processor was applied and you want to rerank documents differently based on an aggregated field. For this rerank type, specify the following fields.
+
+Field  | Data type | Required/Optional | Description
+:--- | :---  | :--- | :--- 
+`by_field` | Object | Required | Provides the rerank processor with reranking information.
+`by_field.target_field` | String | Required |  This can be a path to your field that has a numerical score, in the form of `key[.key]`. 
+`by_field.remove_target_field` | Boolean | Optional | if `true` it will delete the `target_field` used to perform reranking. Otherwise, the field is kept in the search result. Default is `false`.
+`by_field.keep_previous_score` | Boolean | Optional | if `true` it will append a `previous_score`, this was the score calculated before reranking, its useful when debugging. When `false`, there wont be an additional field in your search result. Default is `false`.
+
+## Example 
+
+The following example demonstrates using a search pipeline with a `rerank` processor implemented using the `by_field` rerank type.
+
+### Create an index with data
+Create an index and ingest data that has a numerical score you wish to rerank by. In this example we will ingest the following data for a index named `book-index`
+
+```json
+POST _bulk
+{ "index": { "_index": "book-index", "_id": "1" } }
+{ "title": "The Lost City", "author": "Jane Doe", "genre": "Adventure Fiction", "reviews": { "stars": 4.2 }, "description": "An exhilarating journey through a hidden civilization in the Amazon rainforest." }
+{ "index": { "_index": "book-index", "_id": "2" } }
+{ "title": "Whispers of the Past", "author": "John Smith", "genre": "Historical Mystery", "reviews": { "stars": 4.7 }, "description": "A gripping tale set in Victorian England, unraveling a century-old mystery." }
+{ "index": { "_index": "book-index", "_id": "3" } }
+{ "title": "Starlit Dreams", "author": "Emily Clark", "genre": "Science Fiction", "reviews": { "stars": 4.5 }, "description": "In a future where dreams can be shared, one girl discovers her imagination's power." }
+{ "index": { "_index": "book-index", "_id": "4" } }
+{ "title": "The Enchanted Garden", "author": "Alice Green", "genre": "Fantasy", "reviews": { "stars": 4.8 }, "description": "A magical garden holds the key to a young girl's destiny and friendship." }
+
+```
+{% include copy-curl.html %}
+
+
+### Creating a search pipeline
+
+The following request creates a search pipeline with a `by_field` rerank type response processor. We want to rerank `by_field` using a nsted `target_field` "reviews.stars", we are also interested in understanding how our documents ranked before applying this rerank type so we will enable `keep_previous_score`.
+
```

Comment:
@kolchfa-aws I had to make a couple of assumptions in my rewrites here. Please ensure that they retain technical accuracy.